Lib/enum.py

```diff
@@ -946,3 +942,42 @@ def _decompose(flag, value):
         # we have the breakdown, don't need the value member itself
         members.pop(0)
     return members, not_covered
+
+def global_int_repr(self):
+    return '%s.%s' % (self.__class__.__module__, self.name)
+
+def global_flag_repr(self):
+    module = self.__class__.__module__
+    if self._name_ is not None:
+        return '%s.%s' % (module, self.name)
+    value = self._value_
+    members = []
+    negative = value < 0
+    if negative:
+        value = ~value
+    for m in self.__class__:
+        if value & m._value_:
+            value &= ~m._value_
+            members.append('re.%s' % (m._name_, ))
+    if value:
+        members.append(hex(value))
```

Comment:
For general code I would add a class method for converting value to string. Depending on the class octal or binary may be better than hexadecimal.

Reply:
Fair point.